test/test_listen_socket.cpp

```diff
@@ -203,3 +203,73 @@ TORRENT_TEST(partition_listen_sockets_op_ports)
 	TEST_EQUAL(eps.size(), 2);
 }
 
+bool listen_endpoint_cmp(aux::listen_endpoint_t const& l, aux::listen_endpoint_t const& r)
```

Comment:
you added a comparison operator to ``aux::listen_endpoint_t``. I would expect the test to be able to use that instead of this one. Perhaps it was failing because it wasn't marked ``const`` correctly.